soc: x86: Clean up GPIO related defines

akanisetti opened the pull request:
Clean up and refactor x86 SoC headers in preparation of adding new platforms in the future.

Signed-off-by: Krishna, Anisetti Avinash <anisetti.avinash.krishna@intel.com>

MaureenHelm reviewed (changes requested):
Please extract the whitespace changes to a separate commit from the functional changes.

You also need to rebase and remove the merge commits, and fix the failed compliance check:
```
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

77ec04084d83908f93093b563f27740926ed3be1: author email (Krishna, Anisetti Avinash <anisetti.avinash.krishna@intel.com>) does not follow the syntax: First Last <email>.
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

1: UC2 Signed-off-by: must have a full name
3: UC4 Line exceeds max length (91>75): "Clean up and refactor x86 SoC headers in preparation of adding new platforms in the future."
```

najumon1980 reviewed (changes requested):
Need DTS changes and so please update those changes as well

jhedberg commented on drivers/gpio/gpio_intel.c:
```diff
@@ -569,11 +566,12 @@ int gpio_intel_init(const struct device *dev)
 	isr_devs[nr_isr_devs++] = dev;
 
+#ifdef CONFIG_SOC_APOLLO_LAKE
 	/* route to IRQ 14 */
 
 	sys_bitfield_clear_bit(regs(dev) + REG_MISCCFG,
 			       MISCCFG_IRQ_ROUTE_POS);
-
+#endif
```
Have you verified that this code block was not needed for EHL?

Btw, whenever possible, it's better to use `if (IS_ENABLED(CONFIG_...)) {` instead of `#ifdef`, since it produces more readable code. I think in this case it should be possible to convert the #ifdef to `IS_ENABLED()`

MaureenHelm reviewed (changes requested):
It looks like you changed a bunch of whitespace in one commit and then changed it back in the second commit, which makes it very difficult to see the functional changes. If you're going to change the whitespace, do that (and only that) in one commit, then make your functional change in the second commit. Both commits need meaningful commit messages (currently both have exactly the same commit message).

The alternative is to eliminate all the whitespace changes and only make your functional change in one commit.

MaureenHelm reviewed (commented):
This is much easier to read, thank you!

There's still an open question about impact to EHL https://github.com/zephyrproject-rtos/zephyr/pull/50589#discussion_r980846528

Please also address the failed compliance check:
```
-:30: WARNING:SUSPECT_CODE_INDENT: suspect code indent for conditional statements (8, 8)
#30: FILE: drivers/gpio/gpio_intel.c:569:
+	if(IS_ENABLED(CONFIG_SOC_APOLLO_LAKE)) {
[...]
 	sys_bitfield_clear_bit(regs(dev) + REG_MISCCFG,

-:30: ERROR:SPACING: space required before the open parenthesis '('
#30: FILE: drivers/gpio/gpio_intel.c:569:
+	if(IS_ENABLED(CONFIG_SOC_APOLLO_LAKE)) {

- total: 1 errors, 1 warnings, 52 lines checked
```

akanisetti committed:
soc: x86: Clean up GPIO related defines

Clean up and refactor x86 SoC headers in preparation of adding
new platforms in the future.

Signed-off-by: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>

akanisetti commented:
> This is much easier to read, thank you!
> 
> There's still an open question about impact to EHL [#50589 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/50589#discussion_r980846528)
> 
> Please also address the failed compliance check:
> 
> ```
> -:30: WARNING:SUSPECT_CODE_INDENT: suspect code indent for conditional statements (8, 8)
> #30: FILE: drivers/gpio/gpio_intel.c:569:
> +	if(IS_ENABLED(CONFIG_SOC_APOLLO_LAKE)) {
> [...]
>  	sys_bitfield_clear_bit(regs(dev) + REG_MISCCFG,
> 
> -:30: ERROR:SPACING: space required before the open parenthesis '('
> #30: FILE: drivers/gpio/gpio_intel.c:569:
> +	if(IS_ENABLED(CONFIG_SOC_APOLLO_LAKE)) {
> 
> - total: 1 errors, 1 warnings, 52 lines checked
> ```



> Required



> This is much easier to read, thank you!
> 
> There's still an open question about impact to EHL [#50589 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/50589#discussion_r980846528)
> 
> Please also address the failed compliance check:
> 
> ```
> -:30: WARNING:SUSPECT_CODE_INDENT: suspect code indent for conditional statements (8, 8)
> #30: FILE: drivers/gpio/gpio_intel.c:569:
> +	if(IS_ENABLED(CONFIG_SOC_APOLLO_LAKE)) {
> [...]
>  	sys_bitfield_clear_bit(regs(dev) + REG_MISCCFG,
> 
> -:30: ERROR:SPACING: space required before the open parenthesis '('
> #30: FILE: drivers/gpio/gpio_intel.c:569:
> +	if(IS_ENABLED(CONFIG_SOC_APOLLO_LAKE)) {
> 
> - total: 1 errors, 1 warnings, 52 lines checked
> ```

I have added all the requested changes, could you please review PR with new changes.
